R/weight-streetnet.R

```diff
@@ -1,9 +1,64 @@
-#' bicycle, moped, motorcycle, motorcar, goods, hgv, psv), or a customized
-#' version derived from those.
+#' Weight (or re-weight) an \pkg{sf} or \pkg{silicate} *("SC") formatted OSM
+#' street network according to a named profile, selected from (foot, horse,
+#' wheelchair, bicycle, moped, motorcycle, motorcar, goods, hgv, psv), or a
+#' cusstomized version dervied from those.
+#'
+#' @details The structure of networks generated by this function is dependent
+#' on many aspects of the input network, and in particular on specific
+#' key-value pairs defined in the underlying OpenStreetMap (OSM) data.
+#'
+#' OSM keys which can be used to determine restrictions, and so may be added to
+#' the `keep_cols` argument, include:
+#' \itemize{
+#' \item "access"
+#' \item "bicycle"
+#' \item "foot"
+#' \item "highway"
+#' \item "motorcar"
+#' \item "motor_vehicle"
+#' \item "oneway"
+#' \item "restriction"
+#' \item "toll"
+#' \item "vehicle"
+#' }
+#'
+#' Realistic routing including factors such as access restrictions, turn
+#' penalties, and effects of incline, can only be implemented when the objects
+#' passed to `weight_streetnet` are of \pkg{sc} ("silicate") format, generated
+#' with \link{dodgr_streetnet_sc}. Restrictions applied to ways (in OSMM
+#' terminology) may be controlled by ensuring specific columns are retained in
+#' the `dodgr` network with the `keep_cols` argument. For example, restrictions
+#' on access are generally specified by specifying a value for the key of
+#' "access". Include "access" in `keep_cols` will ensure these values are
+#' retained in the `dodgr` version, from which ways with specified values can
+#' easily be removed or modified, as demonstrated in the examples.
+#'
+#' Restrictions and time-penalties on turns can be implemented by setting
+#' `turn_penalty = TRUE`, which will then honour turn restrictions specified in
+#' OSM (unless the "penalties" table of \link{weighting_profiles} has
+#' `restrictions = FALSE` for a specified `wt_profile`). Resultant graphs are
+#' fundamentally different from the default for distance-based routing. These
+#' graphs may be used directly in most 'dodgr' functions, but generally only if
+#' they have been created by calling this function in the same session, or if
+#' they have been saved and loaded with the \link{dodgr_save_streetnet} and
+#' \link{dodgr_load_streetnet} functions. (This is because the weighted
+#' streetnets also have accompanying data stored in a local temporary cache
+#' directory; attempting to pass a weighted street network without these
+#' accompanying cache files will generally error.)
+#'
+#' Some key-value pairs may also directly influence not just the value of the
+#' graph produced by this function, but also its size. Among these are "oneway"
+#' flags. Without these flags, each edge will be represented in *directed* form,
+#' and so as two rows of the graph for A -> B and B -> A. If a way is tagged in
```

Comment:
"for" or "from"?

Reply:
"for" but obviously need to explain better, which I've now done